### PR TITLE
tfm: Fix nrf54L15 linking with nrf_security targets

### PR DESCRIPTION
**Pull request by Vge0rge**
This fixes an issue for the nRF54L15 when it is build with TFM and the hardware unique key library is also used. The hardware unique key library uses PSA headers which means that we need to make sure that the nrf_security generated headers are being included before the other PSA related headers.

This links the nrf_security target before it links the TFM platform targets.

Ref: NCSDK-29514

test_crypto: PR-672

### CODE REVIEW

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 1]
```diff
@@ -106,6 +110,13 @@ if (${TFM_PARTITION_CRYPTO})
        endif()
    endif()
```
Comment (tomi-font) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:111]: Fix indentation?

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 2]
```diff
@@ -79,20 +79,24 @@ if (${TFM_PARTITION_CRYPTO})
...
 
    if (${CONFIG_HW_UNIQUE_KEY})
      target_compile_definitions(platform_s PUBLIC
        CONFIG_HW_UNIQUE_KEY
        $<$<BOOL:${CONFIG_HW_UNIQUE_KEY_RANDOM}>:CONFIG_HW_UNIQUE_KEY_RANDOM>)
 
-
+       # nrf54l15 uses the PSA headers in the hw_unique_key_cracen.c file, which means
+       # that we need to make sure that the nrf_security PSA headers are included
+       # before any other PSA related headers are included.
```
Comment (tomi-font) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:93]:
```suggestion
       # before any other PSA-related headers.
```

**Review by tomi-font — APPROVED** [review]

**Review by stephen-nordic — APPROVED** [review]

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 3]
```diff
@@ -79,20 +79,24 @@ if (${TFM_PARTITION_CRYPTO})
...
 
-  target_link_libraries(platform_crypto_keys
-    PUBLIC
-      platform_s
-      # Link with tfm_sprt to get the include path for tfm_sp_log.h
-      tfm_sprt
-  )
+  if((NRF_SOC_VARIANT STREQUAL nrf54l15) OR (target STREQUAL nrf54l15))
+  endif()
+
```
Comment (nordicjm) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:84]:
```suggestion
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 4]
```diff
@@ -79,20 +79,24 @@ if (${TFM_PARTITION_CRYPTO})
...
        $<$<BOOL:${CONFIG_HW_UNIQUE_KEY_RANDOM}>:CONFIG_HW_UNIQUE_KEY_RANDOM>)
 
-
+       # nrf54l15 uses the PSA headers in the hw_unique_key_cracen.c file, which means
+       # that we need to make sure that the nrf_security PSA headers are included
+       # before any other PSA related headers are included.
        if((NRF_SOC_VARIANT STREQUAL nrf54l15) OR (target STREQUAL nrf54l15))
+          target_link_libraries(platform_crypto_keys
+           PRIVATE
+              psa_crypto_library_config
```
Comment (nordicjm) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:97]:
```suggestion
           psa_crypto_library_config
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 5]
```diff
@@ -106,6 +110,13 @@ if (${TFM_PARTITION_CRYPTO})
        endif()
    endif()
 
+  target_link_libraries(platform_crypto_keys
+    PUBLIC
+      platform_s
+      # Link with tfm_sprt to get the include path for tfm_sp_log.h
+      tfm_sprt
```
Comment (nordicjm) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:117]:
```suggestion
    PUBLIC
    platform_s
    # Link with tfm_sprt to get the include path for tfm_sp_log.h
    tfm_sprt
```

Keep in-fitting with other parts of the file

**Comment by tomi-font** [comment]
@nordicjm Is it okay that I fix this in my PR (https://github.com/nrfconnect/sdk-nrf/pull/17463) so we don't have to make this re-go through CI?

**Review by tomi-font — APPROVED** [review]

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 6]
```diff
@@ -79,33 +79,41 @@ if (${TFM_PARTITION_CRYPTO})
...
+      CONFIG_HW_UNIQUE_KEY
+      $<$<BOOL:${CONFIG_HW_UNIQUE_KEY_RANDOM}>:CONFIG_HW_UNIQUE_KEY_RANDOM>)
+
+    # nrf54l15 uses the PSA headers in the hw_unique_key_cracen.c file, which means
+    # that we need to make sure that the nrf_security PSA headers are included
+    # before any other PSA-related headers.
+    if((NRF_SOC_VARIANT STREQUAL nrf54l15) OR (target STREQUAL nrf54l15))
+       target_link_libraries(platform_crypto_keys
+        PRIVATE
+        psa_crypto_library_config
```
Comment (nordicjm) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:94]: cmake indent is 2 spaces, not 1

File: modules/trusted-firmware-m/tfm_boards/CMakeLists.txt [hunk 7]
```diff
@@ -79,33 +79,41 @@ if (${TFM_PARTITION_CRYPTO})
...
+
+       target_sources(platform_crypto_keys
+         PRIVATE
+          ${ZEPHYR_NRF_MODULE_DIR}/lib/hw_unique_key/hw_unique_key_cracen.c
+       )
+    else()
+       target_sources(platform_s
+         PRIVATE
+           ${ZEPHYR_NRF_MODULE_DIR}/lib/hw_unique_key/hw_unique_key.c
+           ${ZEPHYR_NRF_MODULE_DIR}/lib/hw_unique_key/hw_unique_key_kmu.c
```
Comment (nordicjm) [inline modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:105]: align the filenames with PRIVATE

Comment (Vge0rge) [reply modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:105]: I saw that, it is like that in all the file though. I will try to fix it file wide then.

Comment (tomi-font) [reply modules/trusted-firmware-m/tfm_boards/CMakeLists.txt:105]: Can we keep formatting refactors for another time?

Comment (Vge0rge) [reply modules/trusted-firmware-m/tfm_boards/CMakeLists.txt]: I just updated the whole file, please double check.

**Review by tomi-font — APPROVED** [review]

**Pull request merged**